src/node/node_state.h

```diff
@@ -398,52 +398,54 @@ namespace ccf
+                                    EndorsementEndpointsConfiguration&
+                                      endpoint_config) {
+        // Note: Node lock is already taken here as this is called back
+        // synchronously with the call to pal::generate_quote
+
+        if (qi.format == QuoteFormat::amd_sev_snp_v1)
+        {
+          if (!config.attestation.snp_endorsements_servers.empty())
           {
-            CCF_ASSERT_FMT(
-              !config.attestation.snp_endorsements_servers.empty(),
+            throw std::runtime_error(
               "One or more SNP endorsements servers must be specified to fetch "
               "the collateral for the attestation");
-            // On SEV-SNP, fetch endorsements from servers if specified
-            quote_endorsements_client =
-              std::make_shared<QuoteEndorsementsClient>(
-                rpcsessions,
-                endpoint_config,
-                [this, qi](std::vector<uint8_t>&& endorsements) {
-                  std::lock_guard<pal::Mutex> guard(lock);
-                  quote_info = qi;
-                  quote_info.endorsements = std::move(endorsements);
-                  try
-                  {
-                    launch_node();
-                  }
-                  catch (const std::exception& e)
-                  {
-                    LOG_FAIL_FMT("{}", e.what());
-                    throw;
-                  }
-                  quote_endorsements_client.reset();
-                });
-
-            quote_endorsements_client->fetch_endorsements();
-            return;
           }
+          // On SEV-SNP, fetch endorsements from servers if specified
+          quote_endorsements_client = std::make_shared<QuoteEndorsementsClient>(
+            rpcsessions,
+            endpoint_config,
+            [this, qi](std::vector<uint8_t>&& endorsements) {
+              std::lock_guard<pal::Mutex> guard(lock);
+              quote_info = qi;
+              quote_info.endorsements = std::move(endorsements);
+              try
+              {
+                launch_node();
+              }
+              catch (const std::exception& e)
+              {
+                LOG_FAIL_FMT("{}", e.what());
+                throw;
+              }
+              quote_endorsements_client.reset();
+            });
 
-          CCF_ASSERT_FMT(
-            (qi.format == QuoteFormat::oe_sgx_v1 && !qi.endorsements.empty()) ||
-              (qi.format != QuoteFormat::oe_sgx_v1 && qi.endorsements.empty()),
-            "SGX quote generation should have already fetched endorsements");
+          quote_endorsements_client->fetch_endorsements();
+          return;
+        }
+
+        CCF_ASSERT_FMT(
+          (qi.format == QuoteFormat::oe_sgx_v1 && !qi.endorsements.empty()) ||
+            (qi.format != QuoteFormat::oe_sgx_v1 && qi.endorsements.empty()),
+          "SGX quote generation should have already fetched endorsements");
```

Comment:
Should this also be promoted from `ASSERT` to `throw`?

Reply:
Yes, might as well.